feat: add `into_id` methods to all menu item types

rhysd opened the pull request:
Fixes #111

This PR implements `into_id` methods to `MenuItem`, `CheckMenuItem`, `PredefinedMenuItem`, and `Submenu`. It moves the menu item into its menu ID.

Since I'm not understanding the policy to add a new method to traits, I'm not sure `into_id` method should be added to `IsMenuItem` trait. If it should not, please let me know. I'll update this branch to move the trait methods to normal methods.

I didn't use `Rc::into_inner` because it was added at Rust 1.70 and Tauri's MSRV is earlier than the version.

amrbashir commented on src/lib.rs:
```diff
@@ -256,6 +256,8 @@ pub unsafe trait IsMenuItem {
 
     fn id(&self) -> &MenuId;
+
+    fn into_id(self) -> MenuId;
```
It is fine to add this to `IsMenuItem` trait but the trait is meant for internal usages only, so you also need to add it to each struct and also add it to `MenuItemKind` enum

amrbashir replied:
Actually, since there isn't any usage for it internally, it is useless to add it here

amrbashir commented on src/items/check.rs:
```diff
@@ -118,3 +127,12 @@ impl CheckMenuItem {
     }
 }
+
+#[test]
+fn test_from_id_and_into_id() {
+    let id = "TEST ID".to_string();
+    let item = CheckMenuItem::with_id(&id, "test", true, true, None);
+    let expected = MenuId(id);
+    assert_eq!(item.id(), &expected);
+    assert_eq!(item.into_id(), expected);
+}
```
move these tests to `src/items/mod.rs`

rhysd commented:
Thank you for the review. I will handle them tomorrow.

rhysd commented:
@amrbashir I fixed your review comments at 111e8a9

- Smaller trait is better. Since `into_id` was unnecessary for any implementation, I removed the method from the `IsMenuItem` trait
- Moved all `into_id` trait methods to each structs
- Added `MenuItemKind::into_id` method
- Gathered all tests into a single test case at `src/items/mod.rs`
